pkg/loop/internal/mercury/common/test/chain_reader.go

```diff
@@ -2,14 +2,57 @@ package mercury_common_test
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/stretchr/testify/assert"
 
 	mercury_types "github.com/smartcontractkit/chainlink-common/pkg/types/mercury"
 )
 
-type StaticMercuryChainReader struct{}
+var ChainReaderImpl = staticMercuryChainReader{
+	staticMercuryChainReaderConfig: staticMercuryChainReaderConfig{
+		latestHeads: []mercury_types.Head{
+			{
+				Hash:   []byte{1},
+				Number: 1,
+			},
+			{
+				Hash:   []byte{2},
+				Number: 2,
+			},
+		},
+	},
+}
+
+type MercuryChainReaderEvaluator interface {
```

Comment:
that makes sense, i'll rework the packaging

nit: i can't define the interface where it's use and do the `var _ iface = x` trick, right, because that would be a circular dependence. i'd need a pkg just for the interfaces

Reply:
You can do the interface check from package x_test files too